src/Data/Array/NonEmpty.purs

```diff
@@ -0,0 +1,327 @@
+module Data.Array.NonEmpty where
+
+import Prelude
+
+import Control.Alt (class Alt)
+import Control.Alternative (class Alternative)
+import Control.Lazy (class Lazy)
+import Control.Monad.Rec.Class (class MonadRec)
+import Data.Array as A
+import Data.Bifunctor (bimap)
+import Data.Eq (class Eq1)
+import Data.Foldable (class Foldable)
+import Data.FoldableWithIndex (class FoldableWithIndex)
+import Data.FunctorWithIndex (class FunctorWithIndex)
+import Data.Maybe (Maybe(..), fromJust)
+import Data.NonEmpty (NonEmpty, (:|))
+import Data.Ord (class Ord1)
+import Data.Traversable (class Traversable)
+import Data.TraversableWithIndex (class TraversableWithIndex)
+import Data.Tuple (Tuple)
+import Data.Unfoldable (class Unfoldable)
+import Partial.Unsafe (unsafePartial)
+
+newtype NonEmptyArray a = NonEmptyArray (Array a)
+
+instance showNonEmptyArray :: Show a => Show (NonEmptyArray a) where
+  show (NonEmptyArray xs) = "(NonEmptyArray " <> show xs <> ")"
+
+derive newtype instance eqNonEmptyArray :: Eq a => Eq (NonEmptyArray a)
+derive newtype instance eq1NonEmptyArray :: Eq1 NonEmptyArray
```

Comment:
I've been compiling with 0.11.7, and it isn't complaining. Will this bug out at runtime?

Reply:
Given
```purescript
data X a = X
derive instance eq1X :: Eq1 X
```
try.purescript.org gives me "Cannot derive a type class instance for Data.Eq.Eq1 X since instances of this type class are not derivable." Is it possible you're using master by accident?

Reply:
Nope. 0.11.7 as installed by Homebrew. Did a clean install, deleted `output` and `bower_components` and it's all compiling fine for me.

I don't know what to say! Is there a commit after 0.11.7 that would have changed this behavior? Or is it possible that the deployed version on Try PureScript is wonky?

Reply:
This was only merged very recently: https://github.com/purescript/purescript/pull/3207

What is the output of your `purs --version`? Is it possible you've got a newer one installed locally to that project which is overriding it? Also can you link me to the homebrew formula you're using? I've checked locally with my 0.11.7 and I get the same thing; "instances of this type class are not derivable".

Reply:
@hdgarrood this is a `derive newtype instance`, so I think it's fine.

Reply:
lmao of course, I'm being silly. thanks